integration/tests/bft/passthrough/suite.go

```diff
@@ -1,177 +1,69 @@
-		nodeConfig := testnet.NewNodeConfig(flow.RoleConsensus,
-			testnet.WithID(nodeID),
-			testnet.WithLogLevel(zerolog.FatalLevel),
-			testnet.WithAdditionalFlag("--required-verification-seal-approvals=1"),
-			testnet.WithAdditionalFlag("--required-construction-seal-approvals=1"),
-			testnet.WithAdditionalFlag(blockRateFlag),
-		)
-		s.nodeConfigs = append(s.nodeConfigs, nodeConfig)
-	}
+	// filter out base suite execution and verification nodes
+	s.NodeConfigs = s.NodeConfigs.Filter(func(n testnet.NodeConfig) bool {
+		if n.Ghost {
+			return true
+		}
+		return n.Role != flow.RoleExecution && n.Role != flow.RoleVerification
+	})
 
 	// generates one corrupted verification node
 	s.verID = unittest.IdentifierFixture()
 	verConfig := testnet.NewNodeConfig(flow.RoleVerification,
 		testnet.WithID(s.verID),
 		testnet.WithLogLevel(zerolog.ErrorLevel),
 		testnet.AsCorrupted())
-	s.nodeConfigs = append(s.nodeConfigs, verConfig)
+	s.NodeConfigs = append(s.NodeConfigs, verConfig)
 
 	// generates two corrupted execution nodes
 	s.exe1ID = unittest.IdentifierFixture()
 	exe1Config := testnet.NewNodeConfig(flow.RoleExecution,
 		testnet.WithID(s.exe1ID),
 		testnet.WithLogLevel(zerolog.ErrorLevel),
 		testnet.AsCorrupted())
-	s.nodeConfigs = append(s.nodeConfigs, exe1Config)
+	s.NodeConfigs = append(s.NodeConfigs, exe1Config)
 
 	s.exe2ID = unittest.IdentifierFixture()
 	exe2Config := testnet.NewNodeConfig(flow.RoleExecution,
 		testnet.WithID(s.exe2ID),
 		testnet.WithLogLevel(zerolog.ErrorLevel),
 		testnet.AsCorrupted())
-	s.nodeConfigs = append(s.nodeConfigs, exe2Config)
-
-	// generates two collection node
-	coll1Config := testnet.NewNodeConfig(flow.RoleCollection,
-		testnet.WithLogLevel(zerolog.FatalLevel),
-		testnet.WithAdditionalFlag(blockRateFlag),
-	)
-	coll2Config := testnet.NewNodeConfig(flow.RoleCollection,
-		testnet.WithLogLevel(zerolog.FatalLevel),
-		testnet.WithAdditionalFlag(blockRateFlag),
-	)
-	s.nodeConfigs = append(s.nodeConfigs, coll1Config, coll2Config)
+	s.NodeConfigs = append(s.NodeConfigs, exe2Config)
 
-	// Ghost Node
-	// the ghost node's objective is to observe the messages exchanged on the
-	// system and decide to terminate the test.
-	// By definition, ghost node is subscribed to all channels.
-	s.ghostID = unittest.IdentifierFixture()
-	ghostConfig := testnet.NewNodeConfig(flow.RoleExecution,
-		testnet.WithID(s.ghostID),
-		testnet.AsGhost(),
-		testnet.WithLogLevel(zerolog.FatalLevel))
-	s.nodeConfigs = append(s.nodeConfigs, ghostConfig)
-
-	// generates, initializes, and starts the Flow network
-	netConfig := testnet.NewNetworkConfig(
+	s.BaseSuite.StartCorruptedNetwork(
 		"bft_passthrough_test",
-		s.nodeConfigs,
-		// set long staking phase to avoid QC/DKG transactions during test run
-		testnet.WithViewsInStakingAuction(10_000),
-		testnet.WithViewsInEpoch(100_000),
+		10_000,
+		100_000,
+		func() insecure.AttackOrchestrator {
+			s.Orchestrator = NewDummyOrchestrator(s.Log)
```

Comment:
The `bft.BaseOrchestrator` is an excellent abstraction that provides a clear and organized structure. To further streamline this process, I suggest refactoring the `DummyOrchestrator` to utilize the `BaseOrchestrator` as a component, leveraging its capabilities and allowing for a more efficient implementation. In essence, `DummyOrchestrator` is very similar to the `BaseOrchestrator`, except that it also tracks certain types of messages.

Reply:
https://github.com/onflow/flow-go/pull/3877/commits/912800fdd5d723ef5ca23a4b6a53fe7e94c1c6c4